### PR TITLE
Computer vision refactoring

### PR DESCRIPTION
**Pull request by Aliaa-Ezz**

### CODE REVIEW

File: computervision/src/main/java/ComputerVision.kt [hunk 1]
```diff
@@ -17,8 +15,8 @@ interface ComputerVision {
     suspend fun setMode(mode: CVInput) //app change mode of glasses
 
     companion object Factory {
-        fun create(sensorPortal: SensorPortal): ComputerVision =
-            ComputerVisionInAction(sensorPortal)
+        fun create(sensorPortal: SensorPortal<Vision, CVInput> , cvTranslator: CVTranslator): ComputerVision =
+            ComputerVisionInAction(sensorPortal , cvTranslator )
```
Comment (mhashim6) [inline computervision/src/main/java/ComputerVision.kt:18]: How about:

```kotlin
fun create(sensorPortal: SensorPortal<Vision, CVInput>, cvTranslator: PortalTranslator<Vision, CVInput> = CVTranslator()): ComputerVision =
            ComputerVisionInAction(sensorPortal, cvTranslator)
```
Give it a default value for the `translator`.

File: computervision/src/main/java/ComputerVision.kt [hunk 2]
```diff
@@ -30,7 +28,7 @@ sealed class Vision : SensorReading {
     class ObjectDetection (val objects: List<DetectedObject>): Vision()
 }
 
-sealed class CVInput : SensorInput {
+sealed class CVInput : SensorControl {
```
Comment (mhashim6) [inline computervision/src/main/java/ComputerVision.kt]: maybe `CVControl` too?

**Review by mhashim6 — CHANGES_REQUESTED** [review]
Bravo! some minor changes and we are done here.
Please write tests for your `CVTranslator`.

File: computervision/src/main/java/ComputerVision.kt (hunk 2 above)
Comment (Aliaa-Ezz) [reply computervision/src/main/java/ComputerVision.kt]: Yes, makes sense 💡

**Review by mhashim6 — CHANGES_REQUESTED** [review]
Very Nice! you installed `Gson` to gradle on your own. A new level, that's why we'll scale up again: your commit should've been split up into two separate commits for clarity. take that into consideration in the next commits. On the other hand, a full test suit is always good news.

**Comment by mhashim6** [comment]
I pushed to `arch_refactoring` branch, please update your branch accordingly. Regarding `SensorPortal<R, C>` thingy.

**Review by mhashim6 — APPROVED** [review]
Done and done.

**Comment by mhashim6** [comment]
@Aliaa-Ezz you should pull request to `arch_refactoring`, not `devel`.

**Comment by mhashim6** [comment]
@Aliaa-Ezz you should pull request to `arch_refactoring`, not `devel`.
Please close this PR and create a new one.

**Pull request closed without merging**